src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -91,41 +103,61 @@ public ActionPlanDTO createActionPlan(
       backoff = @Backoff(delayExpression = "#{${retries.backoff}}"))
   @Override
   public List<ActionPlanDTO> getActionPlansBySelectors(
-      final String collectionExerciseId, final Boolean activeEnrolment) {
+      final String collectionExerciseId, boolean activeEnrolment) {
     log.debug(
         "Retrieving action plan for selectors, " + "collectionExerciseId: {}, activeEnrolment: {}",
         collectionExerciseId,
         activeEnrolment);
 
-    MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
-    queryParams.add("collectionExerciseId", collectionExerciseId);
-    queryParams.add("activeEnrolment", activeEnrolment.toString());
-    UriComponents uriComponents =
+    final MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+    queryParams.add(SELECTOR_COLLECTION_EXERCISE_ID, collectionExerciseId);
+    queryParams.add(SELECTOR_ACTIVE_ENROLMENT, Boolean.toString(activeEnrolment));
+
+    final UriComponents uriComponents =
         restUtility.createUriComponents(appConfig.getActionSvc().getActionPlansPath(), queryParams);
 
-    ResponseEntity<List<ActionPlanDTO>> responseEntity;
-    try {
-      responseEntity =
-          restTemplate.exchange(
-              uriComponents.toString(),
-              HttpMethod.GET,
-              null,
-              new ParameterizedTypeReference<List<ActionPlanDTO>>() {});
-    } catch (HttpClientErrorException e) {
-      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-        return null;
-      }
-      throw e;
-    }
+    final ResponseEntity<List<ActionPlanDTO>> responseEntity =
+        restTemplate.exchange(
+            uriComponents.toString(),
+            HttpMethod.GET,
+            null,
+            new ParameterizedTypeReference<List<ActionPlanDTO>>() {});
 
     log.debug(
         "Successfully retrieved action plan for selectors, "
             + "collectionExerciseId: {}, activeEnrolment: {}",
         collectionExerciseId,
         activeEnrolment);
+
     return responseEntity.getBody();
   }
 
+  @Override
+  public ActionPlanDTO getActionPlanBySelectors(
+      String collectionExerciseId, boolean activeEnrolment) throws CTPException {
+
+    final List<ActionPlanDTO> actionPlans =
+        getActionPlansBySelectors(collectionExerciseId, activeEnrolment);
+
+    if (actionPlans == null) {
+      throw new CTPException(
```

Comment:
How is this ever going to be null? The behaviour of getActionPlansBySelectors has been changed to throw a runtime exception in the event of a non 2XX HTTP response code

Reply:
205 No content is returned when no action plans exist so responseEntity.getBody() would be null.